src/plugins/discover/public/application/apps/main/discover_main_route.tsx

```diff
@@ -140,6 +138,7 @@ export function DiscoverMainRoute({ services, history }: DiscoverMainProps) {
       history={history}
       indexPatternList={indexPatternList}
       savedSearch={savedSearch}
+      savedSearchId={savedSearch.id}
```

Comment:
`DiscoverMainApp` will not be unmounted due to saved search reset, only updated. This check will not be passed, since `indexPattern` and `savedSearch` will be updated at one time. 
```
 if (!indexPattern || !savedSearch) {
    return <LoadingIndicator />;
 }
```